docs/contributing-testing.md

```diff
@@ -0,0 +1,106 @@
+# Testing
+
+This section describes how to run Wasmtime's tests and add new tests.
+
+Before continuing, make sure you can [build
+Wasmtime](./contributing-building.html) successfully. Can't run the tests if you
+can't build it!
+
+## Running All Tests
```

Comment:
Hmm, it seems we’re missing mention about the WASI tests for our WASI impl in the form of `wasi-common` crate. Now I’m happy to merge this as-is and I can add the relevant section afterwards, or we could do it here and now. I could directly push to this branch as well if that’s easier. Let me know what you reckon!

Reply:
Yeah, I haven't really been involved in the WASI work so I'm unfamiliar with it. It would be awesome if you made a follow up PR that documented its tests, and I'll be happy to review!

Reply:
SGTM!